applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h

```diff
@@ -120,20 +120,42 @@ class GeometryUtilities
     {
         KRATOS_TRY;
 
-        // We loop over all nodes and compute the part of the sensitivity which is in direction to the surface normal
+        ProjectNodalVariableOnDirection(rNodalVariable, NORMALIZED_SURFACE_NORMAL);
+
+        KRATOS_CATCH("");
+    }
+
+    void ProjectNodalVariableOnDirection( const Variable<array_3d> &rNodalVariable, const Variable<array_3d> &rNormalVariable)
```

Comment:
If this is a generic function, then also the input variables should be generic, i.e.:

```suggestion
    void ProjectNodalVariableOnDirection( const Variable<array_3d> &rNodalVariable, const Variable<array_3d> &rDirectionVariable)
```